### ISSUE TITLE
scrollObservable question

### ISSUE THREAD

**Issue by calbear47** [tjoskar/ng-lazyload-image#187]
Thank you for such a great component. I've been able to utilize it in so many different scenarios, and it works great. 

However, I've run into a case where I need to load a new photo on the page without re-initializing the page. All the data bindings for the model on the page are using the async pipe, so when the data changes everything changes as expected...except the updated image because it's not loading. 

I've been reading your source code, and I'm trying understand how the pass my own scrollObservable....

My first attempt is somthing like this: 

```js
 @ViewChild('detailContainer', {read: ElementRef}) scrollContainer:ElementRef; // ref to old scollTarget
 lazyLoadImage$ : Subject<ElementRef> = new Subject();
 ngOnInit() {
    this.route.data.shareReplay().subscribe(({facility})=>{
              this.lazyLoadImage$.next(this.scrollContainer); //Every time a new facility is loaded, emit a new value to the scrollObservable
});

```

Then in the template:

```html

<div  class="full-image" [lazyLoad]="'https://facilityimageapi.imgix.net/'+(facility | async)?.thumbnail+'?auto=format&w=600&h=auto&q=60'" [scrollObservable]="lazyLoadImage$"></div>

```

I feel like I'm going about this completely wrong. Could you give me an example of what to pass to scrollObservable. Thanks!

UPDATE:

I think this is closer, but lazy loaded images(except the first one) are being loaded. Here's my code:

```js
 @ViewChild('detailContainer', {read: ElementRef}) scrollContainer:ElementRef; // ref to old scollTarget
 lazyLoadImage$ : Subject = new Subject();

 ngOnInit() {
    this.route.data.shareReplay().subscribe(({facility})=>{
              this.lazyLoadImage$.next(this.scrollContainer); //Every time a new facility is loaded, emit a new value to the scrollObservable
});

  ngAfterViewInit(){
    this.scrollObservable = Observable.merge(Observable.fromEvent(this.scrollContainer.nativeElement, 'scroll'), this.lazyLoadImage$);
  }

```

Then in my template:

```html

<div  class="full-image" [lazyLoad]="'https://facilityimageapi.imgix.net/'+(facility | async)?.thumbnail+'?auto=format&w=600&h=auto&q=60'" [scrollObservable]="scrollObservable"></div>

```

Any ideas why this isn't working....even when I try to force load with Next on my subject?

**Comment by tjoskar** [comment]
Hi @calbear47,

> However, I've run into a case where I need to load a new photo on the page without re-initializing the page.

This is unfortunately not supported. If I understand you correctly, you have a couple of URLs to images that you are loading with the async pipe and you want to lazyload the images when the URL is ready or even replace the image if the URL changes. 

Let's say we have the following template:
```html
<div [lazyLoad]="'https://someimages.com/'+(facility | async)'" [scrollObservable]="scrollObservable"></div>
```

When the template is loaded this directive (ng-lazyload-image) will kick in and check if the image (div) is in viewport, if it is, the directive will try to lazy load the image, which is at this point: 'https://someimages.com/' so we will fail and unsubscribe the (scroll)observable and will not try to load the image again, even though the url changes (facility emits).

This should, however, be quite easy to implement, see my comment here https://github.com/tjoskar/ng-lazyload-image/issues/140#issuecomment-290899163

I maybe will be able to implement it next week but I will gladly accept pull requests :)

Let me know if I completely misunderstood you ;)

**Comment by calbear47** [comment]
Hi @tjoskar,

Thanks for your reply. 

Ok, I will try to give it a shot next week if I have time. I need this functionality for a project, so it's a good opportunity to contribute....but I might need some guidance.

**Comment by tjoskar** [comment]
Hi @calbear47,

Can you update to version `3.3.1` to see if it solves your issue? 
You should not need to include any observable, just:
```html
<div  [lazyLoad]="'https://facilityimageapi.imgix.net/'+(facility | async)?.thumbnail+'?auto=format&w=600&h=auto&q=60'"></div>
```

**Comment by calbear47** [comment]
Hi @tjoskar,

Wow! You're a man of your word! Seriously, I didn't' expect you to actually fix this, but this is a pleasant surprise. 

I did fix my problem..90%. It's workable solution completely, so thank you! If there is one improvement I might suggest, hear me out:

Here is exactly how I'm using it:

```
<div class="image-container">
  <img class="pixel-image" [src]="'https://imageapi.net/'+(facility | async)?.thumbnail+'?auto=format&w=800&h=auto&q=1&blur=50'"  alt="">
  <div class="full-image" [lazyLoad]="'https://imageapi.net/'+(facility | async)?.thumbnail+'?auto=format&w=600&h=auto&q=50'"  [scrollTarget]="detailContainer"></div>
</div>

```

So, when the page initially loads everything works as expected. If the user is on a slow connection, they will first see the `<img>` which is a very very small size,0 blurred image that will load instantly, and then the full image is lazy-loaded in. This behavior worked before your update.

With your update, when the `facility` model changes dynamically on the page, the image does switch now to the current model. So that's a huge improvement. However, ideally I want the user to see the updated blurred image first while the new `facility` image is loaded in(Especially important for mobile users). 

I ran some test, and I think I pointed the problem to the `.ng-lazyloaded` class that is applied to the lazy loaded div once the image is fully loaded. If you removed the `ng-lazyloaded` class when you detect a new image is being loaded, and then re-apply once the image has loaded, this should give the expected behavior I was looking for. 

Thanks again for your work, this is a great library.
